### PR TITLE
Rename bookmarks name

### PR DESCRIPTION
**Pull request by robert00s**
Close: #3226
When hold bookmark we can now change it name.

screenshot 2017-09-29 22-22-46
screenshot 2017-09-29 22-23-18

### CODE REVIEW

File: frontend/apps/reader/modules/readerbookmark.lua [hunk 1]
```diff
@@ -186,7 +188,9 @@ function ReaderBookmark:onShowBookmark()
         if not self.ui.document.info.has_pages then
             page = self.ui.document:getPageFromXPointer(page)
         end
-        v.text = _("Page") .. " " .. page .. " " .. v.notes .. " @ " .. v.datetime
+        if v.text == nil or v.text == "" then
+            v.text = _("Page") .. " " .. page .. " " .. v.notes .. " @ " .. v.datetime
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerbookmark.lua]: This is from before I added the template function. Should we turn it into `Page %1 %2 @ %3` even though that looks a little weird?

**Comment by KenMaltby** [comment]
I notice that you can highlight a paragraph then when you long press on it in "Bookmarks" you get a page, "Bookmark details" that shows the entire highlighted text.  Now if you could add a note here, you could have the much requested note taking feature.

File: frontend/apps/reader/modules/readerbookmark.lua (hunk 1 above)
Comment (robert00s) [reply frontend/apps/reader/modules/readerbookmark.lua]: Of course :)

**Comment by robert00s** [comment]
@KenMaltby 
I'll try to add note functionality in the near future.

**Comment by KenMaltby** [comment]
In a way you have already.  Since the "Name" includes the whole highlighted text, anything you add to the name could be considered a notation.  I don't know what separators are allowed but adding something like" //Note: This is the thing I want to use in the xyz project"  added to the "name" would do the job.  I don't think these are actual file names after all.

**Pull request merged**